refactor: rename ovh-angular-responsive-page-switcher

radireddy opened the pull request:
replace module name from ovh-angular-responsive-page-switcher  to ngOvhResponsivePageSwitcher
BREAKING CHANGE: module is now named ngOvhResponsivePageSwitcher

Commit:
refactor: rename ovh-angular-responsive-page-switcher

replace module name from ovh-angular-responsive-page-switcher  to ngOvhResponsivePageSwitcher
BREAKING CHANGE: module is now named ngOvhResponsivePageSwitcher

antleblanc commented on .npmignore:
```diff
@@ -1,10 +1,2 @@
-.eslintrc.json
-.*.swp
-example
+githubBanner.png
```
We can remove both `githubBanner.png` and `.npmignore` files.

antleblanc commented on README.md:
```diff
@@ -1,78 +1,39 @@
-It detects available width for displaying page in 2 distinct modes :
+# ng-ovh-responsive-page-switcher
 
+> `ng-ovh-responsive-page-switcher` module is a set of 2 directives (`responsive-switch` and `responsive-switch-page`) that enable to display content like pages. It detects available width for displaying page in 2 distinct modes :
```
```suggestion
> It detects available width for displaying page in 2 distinct modes
```

antleblanc commented on README.md:
```diff
@@ -1,78 +1,39 @@
-* [Installation](#installation)
-* [Usage and example](#usage-and-example)
-* [Module Components](#module-components)
+[![Downloads](https://badgen.net/npm/dt/@ovh-ux/ng-ovh-responsive-page-switcher)](https://npmjs.com/package/@ovh-ux/ng-ovh-responsive-page-switcher)[![Dependencies](https://badgen.net/david/dep/ovh-ux/ng-ovh-responsive-page-switcher)](https://npmjs.com/package/@ovh-ux/ng-ovh-responsive-page-switcher?activeTab=dependencies)[![Dev Dependencies](https://badgen.net/david/dev/ovh-ux/ng-ovh-responsive- ...
```
Please move badges right after `> It detects...`

antleblanc commented on README.md:
```diff
@@ -1,78 +1,39 @@
+[![Downloads](https://badgen.net/npm/dt/@ovh-ux/ng-ovh-responsive-page-switcher)](https://npmjs.com/package/@ovh-ux/ng-ovh-responsive-page-switcher)[![Dependencies](https://badgen.net/david/dep/ovh-ux/ng-ovh-responsive-page-switcher)](https://npmjs.com/package/@ovh-ux/ng-ovh-responsive-page-switcher?activeTab=dependencies)[![Dev Dependencies](https://badgen.net/david/dev/ovh-ux/ng-ovh-responsive- ...
+
 
 ## Dependencies
```
Could be renamed as `## Related` and moved after `## Test` part.

antleblanc commented on README.md:
```diff
@@ -103,25 +58,12 @@ The example is now running at `http://localhost:7711/example/`.
 
-See https://github.com/ovh-ux/ovh-angular-responsive-page-switcher/blob/master/LICENSE
+[BSD-3-Clause](LICENSE) © OVH SAS
+
```
Extra empty line.

antleblanc commented on package.json:
```diff
@@ -1,42 +1,54 @@
-  "repository": "ovh-ux/ovh-angular-responsive-page-switcher",
-  "author": "OVH SAS",
+  "keywords": [
+    "angular",
```
keywords aren't related to this package.

We can use the following list:
- page
- responsive
- switcher
- angularjs
- ovhcloud

antleblanc commented on package.json:
```diff
@@ -1,42 +1,54 @@
+    "@commitlint/cli": "^8.3.5",
+    "@commitlint/config-angular": "^8.3.4",
+    "@ovh-ux/component-rollup-config": "^7.0.1",
+    "angular-mocks": "~1.7.9",
```
Is `angular-mocks` used?

radireddy replied:
may be not

antleblanc commented on package.json:
```diff
@@ -1,42 +1,54 @@
+    "@commitlint/config-angular": "^8.3.4",
+    "@ovh-ux/component-rollup-config": "^7.0.1",
+    "angular-mocks": "~1.7.9",
+    "angular-scenario": "~1.7.9",
```
Is `angular-mocks` used?

antleblanc commented on package.json:
```diff
@@ -1,42 +1,54 @@
+    "husky": "^4.2.3",
+    "lint-staged": "^10.1.2",
+    "npm-run-all": "^4.1.5",
+    "ovh-angular-responsive-popover": "~5.0.0-alpha.2"
```
Is `ovh-angular-responsive-popover` used?

radireddy replied:
I did not find any reference for it in the code but it was included previously. Also, below note was there in README.md 
__Note__ : `ng-ovh-responsive-page-switcher` is suitable with [ng-ovh-responsive-popover](https://github.com/ovh-ux/ng-ovh-responsive-popover).

antleblanc commented on package.json:
```diff
@@ -1,42 +1,54 @@
+    "angular-animate": "^1.5.0",
+    "angular-sanitize": "^1.5.0",
+    "jquery": "~2.1.3",
+    "lodash": "~3.3.0",
```
Lodash must be defined as a `devDependencies`in v4.

antleblanc commented on src/index.js:
```diff
@@ -0,0 +1,17 @@
+import 'angular-animate';
+
+import responsiveSwitchPage from './ng-ovh-responsive-page-switcher/directive';
+import responsiveSwitch from './ng-ovh-responsive-switch/directive';
```
```suggestion
import directive from './directive';
```

antleblanc commented on src/index.js:
```diff
@@ -0,0 +1,17 @@
+angular
+  .module(moduleName, ['matchmedia-ng', 'ngAnimate'])
+  .directive('responsiveSwitchPage', responsiveSwitchPage)
+  .directive('responsiveSwitch', responsiveSwitch);
```
```suggestion
  .directive('responsiveSwitch', directive);
```

antleblanc commented on src/index.js:
```diff
@@ -0,0 +1,17 @@
+import 'matchmedia-ng';
+import 'angular-animate';
+
+import responsiveSwitchPage from './ng-ovh-responsive-page-switcher/directive';
```
```suggestion
import pageDirective from './page/directive';
```

antleblanc commented on src/index.js:
```diff
@@ -0,0 +1,17 @@
+import responsiveSwitchPage from './ng-ovh-responsive-page-switcher/directive';
+import responsiveSwitch from './ng-ovh-responsive-switch/directive';
+
+import './ng-ovh-responsive-page-switcher.less';
```
```suggestion
import './index.less';
```

antleblanc commented on package.json:
```diff
@@ -1,42 +1,54 @@
+    "angular": "^1.7.5",
+    "angular-animate": "^1.5.0",
+    "angular-sanitize": "^1.5.0",
+    "jquery": "~2.1.3",
```
Is `jquery` used?

radireddy replied:
it was used, i replaced it with $window

radireddy replied:
maybe it's not required anymore

antleblanc commented on package.json:
```diff
@@ -1,42 +1,53 @@
 {
-  "name": "ovh-angular-responsive-page-switcher",
+  "name": "@ovh-ux/ng-ovh-responsive-page-switcher",
   "version": "1.1.1",
```
```suggestion
  "version": "0.0.0",
```

antleblanc committed:
refactor: rename ovh-angular-responsive-page-switcher

replace module name from ovh-angular-responsive-page-switcher to ngOvhResponsivePageSwitcher

BREAKING CHANGE: module is now named ngOvhResponsivePageSwitcher
